db/init.sql

```diff
@@ -4,13 +4,315 @@ CREATE DATABASE mm_db;
 CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
 
 CREATE TYPE user_role AS ENUM ('admin', 'teacher', 'student');
+CREATE TYPE attempt_state AS ENUM ('in_progress', 'completed');
+-- NOTE(nrydanov): Need to think of other types together
+CREATE TYPE block_type AS ENUM ('assignment', 'text', 'media', 'quiz');
+
 
 CREATE TABLE users (
     id uuid NOT NULL PRIMARY KEY,
+    -- Auth data
     email varchar(128) UNIQUE NOT NULL,
+    password bytea NOT NULL,
+    -- Identity data
     name varchar(32) NOT NULL,
     surname varchar(32) NOT NULL,
     patronymic varchar(32),
-    role user_role NOT NULL,
-    password bytea NOT NULL
+    date_of_birth date,
+    
+    -- University data
+    -- NOTE(nrydanov): This data should be enough to determine exact academic group
+    faculty_id uuid NOT NULL,
+    specialty_id uuid NOT NULL,
+    admission_date date NOT NULL,
+    -- Service info
+    -- NOTE(nrydanov): Can be used to make admin a teacher for a while
+    role_override user_role,
```

Comment:
As I understand it, this field was added in order for teachers to look at their courses from the students' side, but I don't really like the idea of a field that will be used only for teachers, because there will be a lot more students than teachers. If it's the only way to do this, then I agree

In addition to line 29, I believe that administrators should have privileges to view pages, both those that are accessible only to teachers and only to students